kms/src/main/java/io/kroxylicious/kms/service/Kms.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kms.service;
+
+import java.util.concurrent.CompletionStage;
+
+import javax.crypto.SecretKey;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * Abstracts the KMS operations needed for Envelope Encryption
+ * @param <K> The type of Key Encryption Key id.
+ * @param <E> The type of encrypted Data Encryption Key.
+ */
+public interface Kms<K, E> {
+
+    /**
+     * Asynchronously generates a new Data Encryption Key, returning it encrypted with the Key Encryption Key given by {@code kekRef}.
+     * The returned DEK can later be decrypted with {@link Kms#decryptEdek(Object, Object)}.
+     * @param kekRef The key encryption key used to encrypt the generated data encryption key.
+     * @return A completion stage for the wrapped data encryption key.
+     * @throws UnknownKeyException If the kek was not known to this KMS.
+     * @throws InvalidKeyUsageException If the given kek was not intended for key wrapping.
+     * @throws KmsException For other exceptions.
+     */
+    @NonNull
+    CompletionStage<E> generateDek(@NonNull K kekRef);
```

Comment:
As @robobario pointed out to me earlier, we don't actually need this right now, since the `InBandKeyManager` we're using doesn't use this operation. I guess we should remove it.